### PR TITLE
auth LUA: support returning empty set in filterForward

### LINKED ISSUE #12436: Lua Records: Support of empty answers on filterForward()
- Program: Authoritative
 - Issue type: Feature request

### Short description

Return empty answers from  [`filterForward()`](https://doc.powerdns.com/authoritative/lua-records/functions.html#filterForward) for addresses in the `NetmaskGroup` argument, instead of `0.0.0.0` or `::0`.

### Usecase

We'd like to use [`createForward()`](https://doc.powerdns.com/authoritative/lua-records/functions.html#createForward) in conjunction with [`filterForward()`](https://doc.powerdns.com/authoritative/lua-records/functions.html#filterForward). Up to now we have answered queries for addresses outside our networks with an empty answer. [`filterForward()`](https://doc.powerdns.com/authoritative/lua-records/functions.html#filterForward) responds with `0.0.0.0` or `::0` by default in this case. Since we do not want to change the usual behavior of our DNS servers, it would be nice if we could continue do so. 

### Description

In addition, [`filterForward()`](https://doc.powerdns.com/authoritative/lua-records/functions.html#filterForward) should accept `''` (empty string) as the `fallback` argument and it should respond with a empty answer in this case.

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
fixes #12436

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
amended for tidy

File: pdns/lua-record.cc [hunk 1]
```diff
@@ -1061,16 +1061,20 @@ static void setupLuaRecords(LuaContext& lua) // NOLINT(readability-function-cogn
       ComboAddress ca(address);
```
Comment (omoerbeek) [inline pdns/lua-record.cc:1061]: If you add an explicit return type (` -> vector<string>`, you can get rid off all the `vector<string>` in the body, i.e. use `return {address}` or `return {}`.

**Review by omoerbeek — APPROVED** [review]
Looks good, one suggestion you can take or leave.

File: pdns/lua-record.cc (hunk 1 above)
Comment (Habbie) [reply pdns/lua-record.cc:1061]: oh i love it

Comment (Habbie) [reply pdns/lua-record.cc:1061]: done

**Pull request merged**